service/controller/v15/resource/cloudformation/delete.go

```diff
@@ -23,14 +23,6 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 		return microerror.Mask(err)
 	}
 
-	if stackStateToDelete.Status == cloudformation.ResourceStatusUpdateInProgress {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "cannot delete CF stacks due to stack state transitioning")
-
-		// TODO control flow via more proper mechanism via something like the
-		// context like it is done for cancelation already.
-		return microerror.Maskf(deletionMustBeRetriedError, "stack state is transitioning")
-	}
```

Comment:
Again handled in https://github.com/giantswarm/aws-operator/pull/1099/files#diff-6d2b44d389deb52257928547dee86ac0R60

The resource should never reach to this `if`. This should be canceled in `GetCurrentState`.

Reply:
The important thing here is that the delete event is still replayed though. The current state logic just cancels which causes the delete event to succeed and never be replayed now. This is wrong IIUTC. IIRC this was the reason why we added the status to the dispatched structure.

Reply:
Good call, thanks. I think I covered it here https://github.com/giantswarm/aws-operator/pull/1099/commits/b27fecf455fc0245e096c6a983f8ff2f1bf7c590